### PR TITLE
[TTAHUB-1029] Fix auto save error conditions

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Solve issues with incomplete goals not saving properly when the autosave is triggered. 
Also cleanup error message on success, so that the "you had a database error" does not persist.
Change 'you' to 'recipient' on next steps for recipient date label 

Make changes to print preview for approved AR's
- 'Activity Summary' should be sentence case
- Remove user first initial from print preview
- Bold the Objective name
- Fix "if Next Steps has two steps the second step is missing the Date (check for both specialist and recipient)"
- Prevent splitting at header onto next page 

Make changes to Print Preview for Goals

- Remove user first initial from print
- Prevent splitting at header (as above)
- The fields 'Activity Report' and 'Grant Numbers' are blank. Removed grant numbers from objectives.
- Update Objective title and break (spacing)

## How to test
Test activity report auto-saving at each step of the process for the create goals form (no goal, goal with no title, goal with  no date, and so forth for objectives)

Check that the visual bugs above are resolved

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1029

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/HeaderUserMenu.js [hunk 1]
```diff
@@ -142,6 +142,7 @@ function HeaderUserMenu() {
       buttonText={user.name}
       showApplyButton={false}
       direction="left"
+      className="no-print"
```
Comment (AdamAdHocTeam) [inline frontend/src/components/HeaderUserMenu.js:145]: I assume this is the hiding of the initial on print. Not sure if it should be mentioned in the testing steps.

File: frontend/src/components/Navigator/index.js [hunk 2]
```diff
@@ -92,6 +92,16 @@ function Navigator({
   const goalForEditing = watch('goalForEditing');
   const activityRecipientType = watch('activityRecipientType');
   const isGoalsObjectivesPage = page.path === 'goals-objectives';
+  const recipients = watch('activityRecipients');
+  const isRecipientReport = activityRecipientType === 'recipient';
+
+  const grantIds = isRecipientReport ? recipients.map((r) => {
```
Comment (AdamAdHocTeam) [inline frontend/src/components/Navigator/index.js:98]: Why did we need to add this?

Comment (thewatermethod) [reply frontend/src/components/Navigator/index.js:98]: See line 174, needed to be passed up to the API

File: frontend/src/components/HeaderUserMenu.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/components/HeaderUserMenu.js:145]: Whoops, I didn't notice this PR includes some stuff I did yesterday. I'm straightening out my git state locally right now, but I updated the description to more accurately represent what changes this PR includes.

File: frontend/src/components/Navigator/index.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/Navigator/index.js:98]: Was it not working before?

Comment (thewatermethod) [reply frontend/src/components/Navigator/index.js:98]: I added it to resolve an error, yeah

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js [hunk 3]
```diff
@@ -94,6 +94,10 @@ export default function NextStepsRepeater({
 
   const stepType = name === 'specialistNextSteps' ? 'specialist' : 'recipient';
 
+  const dateLabel = (index) => (stepType === 'recipient'
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js:97]: Does this need to change for other entity?

File: frontend/src/pages/RecipientRecord/pages/components/PrintableGoal.js [hunk 4]
```diff
@@ -10,11 +10,11 @@ export default function PrintableGoal({ goal }) {
   const { icon } = STATUSES[key] ? STATUSES[key] : STATUSES['Needs Status'];
 
   return (
-    <div className="ttahub-printable-goal padding-x-3 padding-top-3 padding-bottom-2 margin-top-5">
+    <div className="ttahub-printable-goal padding-x-3 padding-top-3 padding-bottom-2 margin-top-5 no-break-within">
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/RecipientRecord/pages/components/PrintableGoal.js:13]: Thanks for adding this 😎

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js (hunk 3 above)
Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js:99]: [Fixed](https://github.com/HHS/Head-Start-TTADP/pull/1080/commits/19e002a4f15fc8860894a87fa771f030fae311be)

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good. ⚡

**Pull request merged**